lib/handler/mruby.c

```diff
@@ -329,6 +312,25 @@ static h2o_mruby_shared_context_t *get_shared_context(h2o_context_t *ctx)
     return *data;
 }
 
+mrb_value prepare_fibers(h2o_mruby_context_t *ctx)
+{
+    mrb_state *mrb = ctx->shared->mrb;
+
+    h2o_mruby_config_vars_t config = ctx->handler->config;
+    mrb_value conf = mrb_hash_new_capa(mrb, 3);
+    mrb_hash_set(mrb, conf, mrb_symbol_value(mrb_intern_lit(mrb, "code")), mrb_str_new(mrb, config.source.base, config.source.len));
+    mrb_hash_set(mrb, conf, mrb_symbol_value(mrb_intern_lit(mrb, "file")), mrb_str_new(mrb, config.path, strlen(config.path)));
+    mrb_hash_set(mrb, conf, mrb_symbol_value(mrb_intern_lit(mrb, "line")), mrb_fixnum_value(config.lineno));
+
+    /* run code and generate handler */
+    mrb_value result =
+        mrb_funcall(mrb, mrb_obj_value(mrb->kernel_module), "_h2o_prepare_app", 1, conf);
+    h2o_mruby_assert(mrb);
```

Comment:
Well, my point in the [previous comment](https://github.com/h2o/h2o/pull/1173#discussion_r110579168) was that the interface of `prepare_fibers` is designed to return without doing an assertion when the Rack handler fails to initialize. See how it is called in `on_context_init`:

```c
    mrb_value fibers = prepare_fibers(handler_ctx);
    if (mrb->exc != NULL) {
        fprintf(stderr, "mruby raised: %s\n", RSTRING_PTR(mrb_inspect(mrb, mrb_obj_value(mrb->exc))));
```

If we do an assert in `prepare_fiber`, then a user cannot see that the rack handler _the user wrote_ has failed to initialize.